android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/TranslationError.kt

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.translate
+
+/**
+ * The types of translation errors that can occur. Has features for determining telemetry error
+ * names and determining if an error needs to be displayed.
+ *
+ * @param errorName The translation error name. The expected use is for telemetry.
+ * @param displayError Signal to determine if we need to specifically display an error for
+ * this given issue. (Some errors should only silently report telemetry or simply revert to the
+ * prior UI state.)
+ *
+ */
+
+sealed class TranslationError(val errorName: String, val displayError: Boolean) : Throwable() {
+
+    /**
+     * Default error for unexpected issues.
+     *
+     * @param originalThrowable The original throwable that lead us to the unknown error state.
+     */
+    class UnknownError(val originalThrowable: Throwable? = null) : TranslationError("unknown", false)
```

Comment:
Should the `originalThrowable` just be an override for the Throwable's `cause`?

Reply:
Ohh, yes, that makes sense! Do you think all of the errors should get a `cause` in that case?

Reply:
Seemed like I should add it to all of them since it could add value down the road for debugging. 

Added cause as required on `UnknownError`, cause as null on `UnexpectedNull` (since we don't really know how it happened), and cause as a non-default optional on the others, but with the cause specified in the mapper.